frontend/src/common/components/formInputField/FormOwnershipInputField.tsx

```diff
@@ -0,0 +1,241 @@
+import React, {useEffect, useState} from "react";
+
+import {Button, Dialog, IconCrossCircle, IconPlus, IconSearch, Table, TextInput} from "hds-react";
+import {useImmer} from "use-immer";
+
+import {useCreateOwnerMutation} from "../../../app/services";
+import {IOwner} from "../../models";
+import {hitasToast} from "../../utils";
+import QueryStateHandler from "../QueryStateHandler";
+import SaveButton from "../SaveButton";
+import FormInputField, {CommonFormInputFieldProps} from "./FormInputField";
+
+interface FormOwnershipInputFieldProps extends CommonFormInputFieldProps {
+    fieldPath: string;
+    requestedField?: string;
+    queryFunction;
+    relatedModelSearchField: string;
+    getRelatedModelLabel: (unknown) => string;
+    placeholder?: string;
+}
+
+const CreateNewOwner = ({formData, setFormData, error, isLoading, cancelAction, confirmAction}) => {
+    return (
+        <>
+            <Dialog.Content>
+                {error ? (
+                    <div>{`Virhe: ${error}`}</div>
+                ) : (
+                    <>
+                        <FormInputField
+                            inputType="text"
+                            label="Nimi"
+                            fieldPath="name"
+                            formData={formData}
+                            setFormData={setFormData}
+                            error={error}
+                            required
+                        />
+                        <FormInputField
+                            inputType="text"
+                            label="Henkilö- tai Y-tunnus"
+                            fieldPath="identifier"
+                            formData={formData}
+                            setFormData={setFormData}
+                            error={error}
+                            required
+                        />
+                        <FormInputField
+                            inputType="text"
+                            label="Sähköpostiosoite"
+                            fieldPath="email"
+                            formData={formData}
+                            setFormData={setFormData}
+                            error={error}
+                        />
+                    </>
+                )}
+            </Dialog.Content>
+            <Dialog.ActionButtons>
+                <Button
+                    onClick={cancelAction}
+                    variant="secondary"
+                    theme="black"
+                >
+                    Peruuta
+                </Button>
+                {!error && (
+                    <SaveButton
+                        isLoading={isLoading}
+                        onClick={confirmAction}
+                    />
+                )}
+            </Dialog.ActionButtons>
+        </>
+    );
+};
+
+export default function FormOwnershipInputField({
```

Comment:
Not really a fan of each model having it's own component implemented this way. Here we have so much duplicate code vs `FormRelatedModelInputField`, but it's fine for now and let's refactor it into a more reusable component when we have a bigger need for it :)

Reply:
Agreed. A more generic component for adding items to a `FormRelatedModelInputField` might be a good idea.